DOC-10051 - Remove semicolons from query/service requests

sarahlwelton opened the pull request:
Requests against `query/service` can't contain semi-colons without causing errors. 

I went through all of the files listed in the txt file on [DOC-10051](https://issues.couchbase.com/browse/DOC-10051) and removed semi-colons from the appropriate requests. 

Fix version in the ticket was listed as 7.1.1, so raising this against the 7.1 branch. 

Requesting a review from Sitaram as the reporter of the original ticket :)

sarahlwelton committed:
[DOC-10051] Removed semicolons from...

query/service request examples in documentation, as per DOC-10051.

sitaramv commented:
Examples are fixed. Can we say that they can't use semicolon non application json header

sarahlwelton committed:
[DOC-10051) Added example to N1QL error code 1040.

The service returns a 1040 error if a user adds an unescaped semi-colon

in a statement when specifying parameters as form data.

sarahlwelton commented:
> Examples are fixed. Can we say that they can't use semicolon non application json header

Sorry, I'll need more information about where the best spot in the docs is to put this!

sitaramv commented:
https://docs.couchbase.com/server/current/n1ql/n1ql-rest-api/examplesrest.html

Request Content Type

For POST requests, the parameters can be specified in the request body, using one of two content types:
       URL-encoded format (Content-type: application/x-www-form-urlencoded)
       JSON format (Content-type: application/json).
URL-encoded is the default.

For GET requests, the parameters are specified in the request URL in URL-encoded format. Clients should ensure that request parameters are URL-encoded when using the URL-encoded content type. URL-encoded format unescaped semicolon (;) is not valid.

simon-dew commented:
@sitaramv &mdash; please have a look at the related pull request [couchbaselabs/cb-swagger#82](https://github.com/couchbaselabs/cb-swagger/pull/82). Does this cover your comment?

sitaramv commented:
https://github.com/couchbaselabs/cb-swagger/pull/82  covers it

simon-dew commented on modules/n1ql/pages/n1ql-language-reference/curl.adoc:
```diff
@@ -519,7 +519,7 @@ FROM CURL("http://localhost:8093/query/service",
 JOIN `travel-sample`.inventory.airline
 ON KEYS route.airlineid
-LIMIT 4;
+LIMIT 4
```
This semicolon doesn't need to go, because it's part of the outer query, not part of the query within the CURL() function

It doesn't actually matter if you do remove it though; I'd leave it in, for consistency with the other examples in this page

sarahlwelton replied:
Did this as a suggestion commit.

simon-dew commented on modules/n1ql/pages/n1ql-language-reference/n1ql-error-codes.adoc:
```diff
@@ -220,6 +220,8 @@ These errors are related to the service.
 | `Error processing [feature]`
 | Error processing that feature.
 
+For example, if you specify request parameters as form data and include an unescaped semicolon (;) in a statement, the Query Service REST API returns a `1040` error.
```
Is it worth rephrasing to emphasize that this example applies to the REST API?

```suggestion
For example, the Query Service REST API returns this error if you specify request parameters as form data and include an unescaped semicolon (;) in a statement.
```

simon-dew reviewed (changes requested):
Thanks @sarahlwelton. Couple of minor suggestions.

sarahlwelton committed:
Update modules/n1ql/pages/n1ql-language-reference/n1ql-error-codes.adoc

Co-authored-by: Simon Dew <39966290+simon-dew@users.noreply.github.com>

sarahlwelton commented on modules/n1ql/pages/n1ql-language-reference/curl.adoc:
```diff
@@ -519,7 +519,7 @@ FROM CURL("http://localhost:8093/query/service",
 JOIN `travel-sample`.inventory.airline
 ON KEYS route.airlineid
-LIMIT 4;
+LIMIT 4
```
```suggestion
LIMIT 4;
```

sarahlwelton committed:
Update modules/n1ql/pages/n1ql-language-reference/curl.adoc

simon-dew reviewed (approved):
LGTM 👍
